src/api/price.ts

```diff
@@ -5,16 +5,44 @@ import {
     PriceStats,
 } from 'src/antelope/types';
 
+interface CachedPrice {
+    lastFetchTime: number | null,
+    lastPrice: number | null,
+}
 
+const priceCache: { [tokenId: string]: CachedPrice } = {};
 
 export const getCoingeckoUsdPrice = async (
     tokenId: string,
 ): Promise<number> => {
-    const stats: PriceStats = await axios.get(
-        getCoingeckoExchangeStatsUrl(tokenId),
-    );
+    const now = Date.now();
 
-    return stats.data[tokenId].usd;
+    if (priceCache[tokenId] &&
+        priceCache[tokenId].lastFetchTime &&
+        now - (priceCache[tokenId].lastFetchTime as number) < 60 * 1000 &&
+        priceCache[tokenId].lastPrice !== null
+    ) {
+        // If less than a minute has passed since the last fetch, return the cached price.
+        return priceCache[tokenId].lastPrice as number;
+    }
+
+    try {
+        const stats: PriceStats = await axios.get(
+            getCoingeckoExchangeStatsUrl(tokenId),
+        );
+
+        if (stats && stats.status === 200) {
+            const price = stats.data[tokenId].usd;
+            priceCache[tokenId] = { lastFetchTime: now, lastPrice: price };
+            return price;
+        } else {
+            console.error(`Error: received status code ${stats.status} from Coingecko.`);
+            return 0;
+        }
+    } catch (error) {
+        console.error('Error: fetching from Coingecko failed.', error);
+        return 0;
```

Comment:
Will this return price as 0? If apis fail we should not return price 0 but mark it as no price was obtained and show that state to user.

Reply:
What you said is exactly what happens. The price == 0 means "no price", therefore the token price is never set therefore the balance page will show the balance with an icon with a tooltip saying exactly that: "Not reliable fiat value found"